hooks/src/index.js

```diff
@@ -186,31 +186,25 @@ export function useContext(context) {
  */
 let afterPaint = () => {};
 
-/** @type {MessageChannel} */
-let mc;
-
+/**
+ * Pending effects consumer.
+ */
 function onPaint() {
-	mc.port1.postMessage(undefined);
+	afterPaintEffects.forEach(component => {
+		component._afterPaintQueued = false;
+		if (!component._parentDom) return;
+		component.__hooks._pendingEffects.forEach(invokeEffect);
+		component.__hooks._pendingEffects = [];
+	});
+	afterPaintEffects = [];
 }
 
 if (typeof window !== 'undefined') {
-	mc = new MessageChannel();
-
 	afterPaint = (component) => {
 		if (!component._afterPaintQueued && (component._afterPaintQueued = true) && afterPaintEffects.push(component) === 1) {
-			requestAnimationFrame(onPaint);
+			requestAnimationFrame(() => setTimeout(onPaint, 0));
```

Comment:
For a tiny bit more perf we could create this function outside of `afterPaint`.

Reply:
Good point! Changed it. Thank you!